python_modules/libraries/dagster-databricks/dagster_databricks/configs.py

```diff
@@ -0,0 +1,594 @@
+        'for Databricks services). You can set --driver-memory, and --executor-memory to a '
+        'smaller value to leave some room for off-heap usage. '
+        'The --jars, --py-files, --files arguments support DBFS and S3 paths.',
+    )
+
+
+def _define_task():
+    return Field(
+        Selector(
+            {
+                'notebook_task': _define_notebook_task(),
+                'spark_jar_task': _define_spark_jar_task(),
+                'spark_python_task': _define_spark_python_task(),
+                'spark_submit_task': _define_spark_submit_task(),
+            }
+        ),
+        description='The task to run.',
+        is_required=True,
+    )
+
+
+def define_databricks_submit_custom_run_config():
+    fields = _define_submit_run_fields()
+    fields['task'] = _define_task()
+    return Field(Shape(fields=fields), description='Databricks job run configuration')
+
+
+def define_databricks_submit_run_config():
+    return Field(
+        Shape(fields=_define_submit_run_fields()), description='Databricks job run configuration',
+    )
+
+
+def _define_s3_storage_credentials():
+    access_key_key = Field(
+        String,
+        description='The key of a Databricks secret containing the S3 access key ID.',
+        is_required=True,
+    )
+    secret_key_key = Field(
+        String,
+        description='The key of a Databricks secret containing the S3 secret access key.',
+        is_required=True,
+    )
+    return Field(
+        Shape(fields={'access_key_key': access_key_key, 'secret_key_key': secret_key_key}),
+        description='S3 storage secret configuration',
+    )
+
+
+def _define_adls2_storage_credentials():
+    storage_account_name = Field(
+        String,
+        description='The name of the storage account used to access data.',
+        is_required=True,
+    )
+    storage_account_key_key = Field(
+        String,
+        description='The key of a Databricks secret containing the storage account secret key.',
+        is_required=True,
+    )
+    return Field(
+        Shape(
+            fields={
+                'storage_account_name': storage_account_name,
+                'storage_account_key_key': storage_account_key_key,
+            }
+        ),
+        description='ADLS2 storage secret configuration',
+    )
+
+
+def _define_storage_credentials():
+    return Selector(
+        {'s3': _define_s3_storage_credentials()},
+        # TODO: uncomment when dagster-azure is merged.
+        # {'s3': _define_s3_storage_credentials(), 'adls2': _define_adls2_storage_credentials()},
+    )
+
+
+def define_databricks_storage_config():
+    secret_scope = Field(
+        String,
+        description='The Databricks secret scope containing the storage secrets.',
+        is_required=True,
+    )
+    return Field(
+        Shape(fields={'secret_scope': secret_scope, 'credentials': _define_storage_credentials()}),
```

Comment:
What do you think about putting `secret_scope` under `credentials`?

Reply:
Sounds good, should simplify the config a little.